File: Frontend/src/data/UserProject.ts
interface UserProject
{
    Id: string,
    UserId: string,
    ProjectId: string
}

export default UserProject

/*
        [Key]
        public string Id {  get; set; }

        // Foreign key
        public string UserId {  get; set; }

        // Foreign key
        public string ProjectId {  get; set; }

        [ForeignKey("UserId")]
        public User User { get; set; }

        [ForeignKey("ProjectId")]
        public Project Project {  get; set; }
*/

File: Backend/Data/Model/UserProject.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Data.Model
{
    // M-to-N relationship between User and Projects table
    public class UserProject
    {

        // Foreign key (composite)
        [ForeignKey("User")]
        public string UserId {  get; set; }

        // Foreign key (composite)
        [ForeignKey("Project")]
        public string ProjectId {  get; set; }

        // Navigation properties for tables
        public User User { get; set; }
        public Project Project {  get; set; }
    }
}
